packages/apps/api/src/handlers/auth.ts

```diff
@@ -1,31 +1,27 @@
-import { APIResponse, isNonBlank, ResponseStatus } from '@weco/identity-common';
-import SierraClient from '@weco/sierra-client';
-import { PatronRecord } from "@weco/sierra-client/lib/patron";
+import { isNonBlank, ResponseStatus } from '@weco/identity-common';
 import { Request, Response } from 'express';
 import { toMessage } from '../models/common';
+import Auth0Client from "@weco/auth0-client";
 
-export async function validateCredentials(sierraClient: SierraClient, request: Request, response: Response): Promise<void> {
-
-  const email = request.body.email;
-  const password = request.body.password;
+export async function validateCredentials(auth0Client: Auth0Client, request: Request, response: Response) {
+  const email: string = request.body.email;
+  const password: string = request.body.password;
 
   if (!isNonBlank(email) || !isNonBlank(password)) {
     response.status(400).json(toMessage("All fields must be provided and non-blank"));
+    return;
+  }
+
+  const userFetchResult = await auth0Client.getUserByEmail(email);
+  if (userFetchResult.status != ResponseStatus.Success) {
+    response.status(404).json(toMessage(userFetchResult.message));
+    return;
   }
 
-  const sierraGet: APIResponse<PatronRecord> = await sierraClient.getPatronRecordByEmail(email);
-  if (sierraGet.status === ResponseStatus.Success) {
-    const sierraValidate: APIResponse<{}> = await sierraClient.validateCredentials(sierraGet.result.barcode, password);
-    if (sierraValidate.status === ResponseStatus.Success) {
-      response.status(200).end();
-    } else if (sierraValidate.status === ResponseStatus.InvalidCredentials) {
-      response.status(401).json(toMessage(sierraValidate.message));
-    } else {
-      response.status(500).json(toMessage(sierraValidate.message));
-    }
-  } else if (sierraGet.status === ResponseStatus.NotFound) {
-    response.status(404).json(toMessage(sierraGet.message))
+  const validationResult = await auth0Client.validateUserCredentials(email, password);
+  if (validationResult.status == ResponseStatus.Success) {
```

Comment:
I think we should be a bit more specific here with the error handling.

The call to `validateUserCredentials(email, password)` should return either `200`, `401` or `500` depending on whether it was successful, the authentication attempt was explicitly denied, or some other unhandled error occurred (could be errors with the Auth0 API). These should be checked for and the respective error return.